src/dotnet/ToolPackage/CommandSettingsRetriver.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.DotNet.Tools;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal static class CommandSettingsRetriver
+    {
+        public static IReadOnlyList<CommandSettings> GetCommands(
+            string packageId,
+            DirectoryPath assetJsonDirectory,
+            bool commandsInNuGetCache)
+        {
+            const string AssetsFileName = "project.assets.json";
+            const string ToolSettingsFileName = "DotnetToolSettings.xml";
+
+            try
+            {
+                var commands = new List<CommandSettings>();
+                var lockFile = new LockFileFormat().Read(assetJsonDirectory.WithFile(AssetsFileName).Value);
+
+                DirectoryPath packageDirectory;
+                if (commandsInNuGetCache)
+                {
+                    packageDirectory =
+                        new DirectoryPath(lockFile.PackageFolders[0]
+                            .Path); //todo no checkin, why there is more than one?
```

Comment:
Get an answer, it is safe to do so. I will add comment

Reply:
Do we need a bounds check on what NuGet is returning?